cmake/BLTPrivateMacros.cmake

```diff
@@ -555,18 +555,19 @@ endmacro(blt_add_hip_executable)
 ##                                    C_LIST <list name>
 ##                                    Fortran_LIST <list name>
 ##                                    Python_LIST <list name>)
+##                                    CMAKE_LIST <list name>)
 ##
-## Filters source list by file extension into C/C++, Fortran, and
-## Python source lists based on the global BLT variables
-## BLT_C_FILE_EXTS, BLT_Fortran_FILE_EXTS, and BLT_Python_FILE_EXTS,
-## respectively.  Files with no extension or generator expressions
-## that are not object libraries (of the form
+## Filters source list by file extension into C/C++, Fortran, Python, and
+## CMake source lists based on BLT_C_FILE_EXTS, BLT_Fortran_FILE_EXTS,
+## and BLT_CMAKE_FILE_EXTS (global BLT variables). Files named
+## "CMakeLists.txt" are also filtered here. Files with no extension
+## or generator expressions that are not object libraries (of the form
 ## "$<TARGET_OBJECTS:nameofobjectlibrary>") will throw fatal errors.
 ## ------------------------------------------------------------------------------
 macro(blt_split_source_list_by_language)
 
     set(options)
-    set(singleValueArgs C_LIST Fortran_LIST Python_LIST)
+    set(singleValueArgs C_LIST Fortran_LIST Python_LIST CMAKE_LIST)
```

Comment:
@kennyweiss I thought about that possibility also. CMake has a lot of settings controlled by variables that start with the substring `CMAKE_`, so I chose a name consistent with that convention rather than one consistent with the convention established by `Fortran_LIST` and `Python_LIST`. 

CMake itself is inconsistent with respect to Python; compare the convention used in https://github.com/Kitware/CMake/blob/master/Modules/FindPythonInterp.cmake (Python is in all caps) to https://github.com/Kitware/CMake/blob/master/Modules/FindPython.cmake (Python is in the usual case).